mleap-spark/src/main/scala/org/apache/spark/ml/bundle/ops/feature/BinarizerOp.scala

```diff
@@ -22,25 +24,36 @@ class BinarizerOp extends SimpleSparkOp[Binarizer] {
       assert(context.context.dataset.isDefined, BundleHelper.sampleDataframeMessage(klazz))
 
       val dataset = context.context.dataset.get
-      model.withValue("threshold", Value.double(obj.getThreshold)).
-        withValue("input_shapes", Value.dataShape(sparkToMleapDataShape(dataset.schema(obj.getInputCol), dataset)))
+      var result = {
+        ParamValidators.checkSingleVsMultiColumnParams(obj, Seq(obj.inputCol), Seq(obj.inputCols))
+        if(obj.isSet(obj.inputCols)) {
+          model.withValue("input_shapes", Value.dataShape(ListShape()))
```

Comment:
This is the only one I encountered at least

Reply:
there are a few other transformers that use `input_shapes, for e.g. `VectorAssembler` etc.

can we do `model.withValue("input_shapes", Value.dataShape(sparkToMleapDataShape(dataset.schema(obj.getInputCols), dataset)))` and let sparkToMleapDataShape to derive the right shape?

Reply:
iirc. sparkToMleapDataShape didn't work with the complex type so I had to special case things out.  But I'll take another look.

Reply:
Right, looking again, the issue is that `sparkToMleapDataShape` expects a `spark.sql.types.StructField`  But we have multiple fields, so`dataset.schema(obj.getInputCols)` returns a `StructType` instead.  I thought about making `sparkToMleapDataShape` work with a StructType, but this ended up being rather complex and had a lot of downstream implications.

The way that VectorAssembler works is to create a DataShape for each of the inputcols, and then serialize a `dataShapeList` for the bundle.  This makes a lot of sense for VectorAssembler since it will always have an `inputCols` param (there is no option in spark for it to be single input format).  Doing that kind of approach might make more sense here too, though will mean we have to change the parameter name since the single vs multi format types will be different.  That seems acceptable though imo?